tests/DOMNodeComparatorTest.php

```diff
@@ -118,6 +126,15 @@ public function assertEqualsFailsProvider()
                 $this->createDOMDocument('<root> bar </root>'),
                 $this->createDOMDocument('<root> BAR </root>'),
             ],
+            [
+                $this->createDOMDocument(
+                    '<!DOCTYPE foo SYSTEM "foo.dtd" [<!ENTITY ns_foo "http://uri.tld/foo">]><i:foo xmlns:i="&ns_foo;"/>'
+                ),
+                $this->createDOMDocument(
+                    '<!DOCTYPE foo SYSTEM "foo.dtd" [<!ENTITY ns_foo "http://uri.tld/foo">]>' .
+                    '<i:foo xmlns:i="&ns_foo;" bar="I am different"/>'
+                ),
+            ],
```

Comment:
While we certainly need these additions, I'm not convinced they're at the correct spot: It's not testing for un-equal. It's testing the recovery (or lack there of) from an implementation related issue.

Shouldn't that be its own test? And thus it's own data provider if need be? Not sure, just pondering...

Reply:
Since `C14N` can also return an empty string, I agree that samples making `C14N()` fail should be on their own providers.

Reply:
Moved cases to its own providers

I did something I dont like: calling a test inside a test. But I think is better than write exactly the same test twice.

I think this can be improved by extracting the private function `nodeToText` to its own internal class, I can do that and provide better testing understanding. That would be out of the scope of this PR, isn't it?